cmd/containerd-shim-runc-v2/task/service.go

```diff
@@ -651,9 +667,33 @@ func (s *service) send(evt interface{}) {
 	s.events <- evt
 }
 
+func (s *service) handleProcessExitL(e runcC.Exit, c *runc.Container, p process.Process) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	s.handleProcessExit(e, c, p)
+}
+
 // s.mu must be locked when calling handleProcessExit
 func (s *service) handleProcessExit(e runcC.Exit, c *runc.Container, p process.Process) {
 	if ip, ok := p.(*process.Init); ok {
+		s.pendingExecsLock.Lock()
+		wg, pendingExecs := s.pendingExecs[p.Pid()]
+		s.pendingExecsLock.Unlock()
+		if pendingExecs {
+			// This is the init process and there are pending execs to process, so
+			// wait until all pending execs are processed before we process our exit.
+			// see: https://github.com/containerd/containerd/issues/9719
+			go func() {
+				wg.Wait()
+				s.handleProcessExitL(e, c, p)
+
+				s.pendingExecsLock.Lock()
+				delete(s.pendingExecs, p.Pid())
```

Comment:
What is the delete done after the `s.handleProcessExitL`? If the pid is still in `s.pendingExecs`, what stopping this condition from getting hit multiple times?